rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -257,6 +257,76 @@ object ProcNormalizeMatcher {
         )
       }
 
+      case p: PInput => {
+        import scala.collection.JavaConverters._
+        // To handle the most common case where we can sort the binds because
+        // they're from different sources, Each channel read starts its free variables at 0.
+        // We check for overlap at the end before sorting.
```

Comment:
I would replace this with

        // To handle the most common case where we can sort the binds because
        // they're from different sources, each channel read (pattern) starts its free variables at 0.
        // We check for overlap at the end after sorting.

In other words, clarify that by "channel read" its the pattern and change "before" to "after".

Reply:
Put it's not a pattern. It's a list of channels. I'll try and re-phrase.